### ISSUE TITLE
@Inherited annotations aren't detected

### ISSUE THREAD

**Issue by doodla** [linkedin/dex-test-parser#26]
My annotation classes look like this - 

```kotlin

import java.lang.annotation.Inherited

@Target(AnnotationTarget.CLASS)
@Inherited
internal annotation class ScreenshotTest

@Target(AnnotationTarget.CLASS)
internal annotation class EspressoTest

```

For a test structure like this -

```kotlin

@ScreenshotTest
abstract class BaseScreenshotTest

class ScreenshotTestA : BaseScreenshotTest


abstract class BaseEspressoTest

@EspressoTest
class EspressoTestA : BaseEspressoTest

@EspressoTest
class EspressoTestA : BaseEspressoTest

```

Using `adb shell am instrument -w -e annotation 'com.company.mobile.ScreenshotTest` works as expected and picks up `ScreenshotTestA`.

However, when using dex-parser, the `@Test` methods in `ScreenshotTestA` don't have the `@ScreenshotTest` annotation.

To get around it, instead of only having the annotation on a base class, we have to do away with the `@Inherited` annotation and manually annotate every class, like in the `@EspressoTest` example.

It would be amazing if dex parser could also detect and return the `@Inherited` annotations along with those on the class.

**Comment by winterDroid** [comment]
yeah that would be awesome!

**Comment by hongbinh** [comment]
We will look into this in more details next quarter, thanks for your patience.

**Comment by hongbinh** [comment]
Fixed in https://github.com/linkedin/dex-test-parser/pull/27

**Comment by doodla** [comment]
#26 was supposedly fixed in #27 but it doesn't look like the fix worked.

I've attached a sample Android project and a parsing test project which verifies that the annotations aren't parsed by `dex-test-parser`. If I'm doing something wrong, please let me know. This issue should be reopened in the meanwhile.

@hongbinh @Malinskiy 

[DexParserSample.zip](https://github.com/linkedin/dex-test-parser/files/4136774/DexParserSample.zip)
[DexTest.zip](https://github.com/linkedin/dex-test-parser/files/4136775/DexTest.zip)


The Android structure is -

```kt
@Inherited
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
annotation class ScreenshotTest

@Inherited
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
annotation class EspressoTest


@RunWith(AndroidJUnit4::class)
abstract class UiTest {
}


@EspressoTest
abstract class EspressoTestBaseClass : UiTest() {
}


@ScreenshotTest
abstract class ScreenshotTestBaseClass : UiTest() {
}


class Test1 : EspressoTestBaseClass() {

    @Test
    fun someTest() {

    }
}
```

Dex parser is able to correctly see `someTest()`, but the only annotation it applies to it is `@Test` and not also `@EspressoTest`.

**Comment by doodla** [comment]
Some additional info.

```kt
@EspressoTest
class Test1 : EspressoTestBaseClass() {

    @Test
    fun someTest() {

    }
}
```
Annotating a class like this makes the `EspressoTest` annotation show up, and the `inherited` property is `true`. But it's useless at this point, unfortunately, because the whole point of the `@Inherited` annotation is so I wouldn't have to annotate it on subclasses.

![image](https://user-images.githubusercontent.com/3683106/73500539-d1e77e80-4377-11ea-8e5f-c094d10705d1.png)
